Semigroup-Monoid.md

```diff
@@ -0,0 +1,318 @@
+As you can see here, we can apply it for both a `Double` and a `String` type, but remember you can use it with many other types.
+
+
+```kotlin
+with(Double.semigroup()){
+    4.5.combine(7.0).combine(2.5)
+}
+//14.0
+```
+
+```kotlin
+with(String.semigroup()){
+    "This is ".combine("functional ") + ("programming.")
+}
+//This is functional programming.
+```
+
+Additionaly `Semigroup` adds `+` syntax to all types for which a `Semigroup` instance exists
+
+^ This is an example which explain the main characteristic about Semigroup´s combine method in a clear way.
+^ We are applying it to a Double and a String in this case, but remember you can use it with many other types.
+^ Additionaly Semigroup adds + syntax (mark +) to all types for which a Semigroup instance exists
+^ With a semigroup, we gain the capability to combine, and that is the main value of it.
+
+---
+
+# Semigroup :: Laws
+
+All Typeclasses __must satisfy some mathematical laws__ in order to be considered a Typeclass.
+
+Arrow has tests in place for those laws so typeclasses can keep their integrity over time.
+
+^ When we think about typeclasses, we assume they need to satisfy some mathematical laws in order to be considered a typeclass.
+^ Those laws are coded and enforced by tests inside Arrow, so we can ensure typeclasses integrity over time.
+^ Some laws could be identity, associativity, composition, and so on.
+
+---
+
+# Semigroup :: Laws
+
+Semigroup satisfies the following law:
+
+- __Associativity__: When adding or multiplying, it doesn't matter how we combine the group of numbers.
+
+```kotlin
+    (a.combine(b)).combine(c)
+```
+ must be the same as: 
+ 
+```kotlin
+    (a.combine(b.combine(c))
+```
+ 
+^ Semigroup satisfies one law: Associativity.
+^ The combine function of Semigroup must be associative, which is to say, in pseudo code: "(a.combine(b)).combine(c)" is equivalent to "a.combine(b.combine(c))".
+
+---
+
+# Monoid
+
+__`Monoid`__ is available in the __arrow-typeclasses__ module.
+
+```groovy
+// gradle
+compile 'io.arrow-kt:arrow-typeclasses:$arrowVersion'
+
+import arrow.typeclasses.Monoid
+```
+^ At this point, we could talk about Monoid type class.
+^ Monoid is provided on the arrow-typeclasses artifact.
+^ You can use it as soon as you add it to your build.gradle file.
+
+---
+
+# Monoid
+
+`Monoid` is a __Typeclass__ which extends the power of `Semigroup` providing an additional `empty` method, to semigroup´s `combine`.
+
+```
+interface Monoid<A> : Semigroup<A> {
+    //...
+}
+```
+
+^ Monoid extends the semigroup type class, adding an empty method to semigroup´s combine.
+
+---
+
+# Monoid :: empty
+
+Having an __empty__ defined allows us to combine all the elements of an potentially empty collection of `T` for which a `Monoid <T>` is defined and as a result we would get a `T` element instead of an `Option <T>`, since we have a specific value to fall back to.
```

Comment:
Specify this is an example, empty is used in other places beside collection combinations

Reply:
I will change this example for a clearer one